src/test/java/itest/MessagingServerIT.java

```diff
@@ -0,0 +1,113 @@
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.http.WebSocket;
+import io.vertx.core.json.JsonObject;
+import itest.bases.StandardSelfTest;
+import itest.util.Utils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+public class MessagingServerIT extends StandardSelfTest {
+
+    public static final String DEPRECATED_COMMAND_PATH = "/api/v1/command";
+
+    public static CompletableFuture<JsonObject> sendMessage(String command, String... args)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        CompletableFuture<JsonObject> future = new CompletableFuture<>();
+
+        Utils.HTTP_CLIENT.webSocket(
+                DEPRECATED_COMMAND_PATH,
+                ar -> {
+                    if (ar.failed()) {
+                        future.completeExceptionally(ar.cause());
+                        return;
+                    }
+                    WebSocket ws = ar.result();
+
+                    ws.handler(
+                            m -> {
+                                JsonObject resp = m.toJsonObject();
+                                String commandName = resp.getString("commandName");
```

Comment:
I think we can simplify this test setup a bit. We don't actually expect the messaging server to act upon the command request, so there isn't really a need to send it a well-formed command message and check that we got a well-formed response that matches what we requested, right? The connection rejection will happen before any of that occurs, so the test case doesn't need to be set up to send nice messages or process responses.

Reply:
Sounds good, so for future reference, tests should only consist of what we actually expect them to execute and not the full range of possibilities?

Reply:
Well, there's a balance. The full range of possibilities is almost literally any behaviour, after all, but in the test cases you wrote, you (very reasonably) handled a lot of those with "if any error occurs, just throw the exception and move on", for example. In this case, what we're interested in is that we get a 410 response, and any other behaviour is incorrect from the point of view of this test. There isn't too much need to be specific about behaviours that are unexpected/incorrect - so long as the behaviour is reported with enough detail to debug the test failure. In this particular example, the test just failing and saying "I expect a 410 response but instead, I got a 200" (or whatever else) seems sufficient - no need to go into "I expect a 410 response, but got a 200, but also the 200 was for the correct command response"

Reply:
Ah makes sense, thank you.